sharedresource/v1alpha1/types_share.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The OpenShift authors.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Share is the Schema for the shares API
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:subresource:status
+// +openshift:compatibility-gen:level=4
+type Share struct {
```

Comment:
Share-what?  or Share-what-for?  Did you brainstorm other more specific names?  I'm not strictly a no, but all sorts of content is shared.  Indicating what it is used for could help.

Reply:
> Share-what? or Share-what-for? Did you brainstorm other more specific names? 

Short answer, yes, there was a brainstorming session with multiple folks, including various "architects".  But @adambkaplan drove the selection for this current name, so unless he has already reached out to you out of band from this PR, I'll defer to him on spelling out an pertinent details around how the latest naming forays went.

> I'm not strictly a no, but all sorts of content is shared. Indicating what it is used for could help.

Reply:
Share kind of suggests Windows "share" volume :-)

Reply:
We went with "Share" after bikeshedding this a bit in the original enhancement proposal.

See the [EP](https://github.com/openshift/enhancements/blob/master/enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md#end-to-end-flow-from-creation-to-opt-in-to-validation-and-then-injections) and discussions in this [PR](https://github.com/openshift/enhancements/pull/318)